cmd/create/ocmrole/cmd.go

```diff
@@ -0,0 +1,387 @@
+			Default:  permissionsBoundary,
+			Validators: []interactive.Validator{
+				aws.ARNValidator,
+			},
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid policy ARN for permissions boundary: %s", err)
+			os.Exit(1)
+		}
+	}
+	if permissionsBoundary != "" {
+		_, err := arn.Parse(permissionsBoundary)
+		if err != nil {
+			reporter.Errorf("Expected a valid policy ARN for permissions boundary: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	if interactive.Enabled() {
+		mode, err = interactive.GetOption(interactive.Input{
+			Question: "Role creation mode",
+			Help:     cmd.Flags().Lookup("mode").Usage,
+			Default:  aws.ModeAuto,
+			Options:  aws.Modes,
+			Required: true,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid role creation mode: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	// Get current OCM account:
+	currentAccount, err := ocmClient.GetCurrentAccount()
+	if err != nil {
+		reporter.Errorf("Failed to get current account: %s", err)
+		os.Exit(1)
+	}
+
+	switch mode {
+	case aws.ModeAuto:
+		reporter.Infof("Creating role using '%s'", creator.ARN)
+		roleARN, err := createRoles(reporter, awsClient, prefix, permissionsBoundary, creator.AccountID, env)
+		if err != nil {
+			reporter.Errorf("There was an error creating the ocm role: %s", err)
+			ocmClient.LogEvent("ROSACreateOCMRoleModeAuto", map[string]string{
+				ocm.Response: ocm.Failure,
+			})
+			os.Exit(1)
+		}
+		reporter.Infof("To create a cluster with these role, run the following command:\n" +
+			"rosa create cluster --sts")
+		ocmClient.LogEvent("ROSACreateOCMRoleModeAuto", map[string]string{
+			ocm.Response: ocm.Success,
+		})
+		err = ocmrole.Cmd.RunE(cmd, []string{roleARN})
+		if err != nil {
+			reporter.Errorf("Unable to link role arn '%s' with the account id : '%s' : %v",
+				roleARN, currentAccount.ID(), err)
+		}
+	case aws.ModeManual:
+		ocmClient.LogEvent("ROSACreateOCMRoleModeManual", map[string]string{})
+		err = aws.GeneratePolicyFiles(reporter, env)
+		if err != nil {
+			reporter.Errorf("There was an error generating the policy files: %s", err)
+			ocmClient.LogEvent("ROSACreateOCMRoleModeManual", map[string]string{
+				ocm.Response: ocm.Failure,
+			})
+			os.Exit(1)
+		}
+		if reporter.IsTerminal() {
+			reporter.Infof("All policy files saved to the current directory")
+			reporter.Infof("Run the following commands to create the ocm role and policies:\n")
+		}
+		commands := buildCommands(prefix, permissionsBoundary, creator.AccountID)
+		fmt.Println(commands)
+	default:
+		reporter.Errorf("Invalid mode. Allowed values are %s", aws.Modes)
+		os.Exit(1)
+	}
+}
+
+func buildCommands(prefix string, permissionsBoundary string, accountID string) string {
+	commands := []string{}
+	name := aws.GetRoleName(prefix, aws.OCMRole)
+	policyName := fmt.Sprintf("%s-Policy", name)
+	iamTags := fmt.Sprintf(
+		"Key=%s,Value=%s Key=%s,Value=%s",
+		tags.RolePrefix, prefix,
+		tags.RoleType, aws.OCMRole,
```

Comment:
Since the role is environment-dependent, we should add the env (integration/staging/production) as a tag

Reply:
added